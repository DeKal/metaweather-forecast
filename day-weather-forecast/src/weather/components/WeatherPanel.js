import React from 'react'
import PropTypes from 'prop-types'
import { Grid } from '@material-ui/core'
import WeatherTile from './WeatherTile'
import Loading from './Loading'

const WeatherPanel = ({ weathers, loading }) => {
  if (loading) {
    return <Loading />
  }
  if (!weathers.length) {
    return 'No weather to display. Please search for a city.'
  }

  return (
    <Grid container>
      {weathers.map((weather) => (
        <Grid key={weather.date} item lg={12 / weathers.length} md={4} xs={6}>
          <WeatherTile
            date={weather.date}
            minTemperature={weather.minTemperature}
            maxTemperature={weather.maxTemperature}
          />
        </Grid>
      ))}
    </Grid>
  )
}

export default WeatherPanel
WeatherPanel.propTypes = {
  weathers: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string,
      minTemperature: PropTypes.number,
      maxTemperature: PropTypes.number
    })
  ),
  loading: PropTypes.bool
}
WeatherPanel.defaultProps = {
  weathers: [],
  loading: false
}
