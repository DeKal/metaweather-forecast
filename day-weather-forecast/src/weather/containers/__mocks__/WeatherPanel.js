import React from 'react'

const MockWeatherPanel = () => <div> Mock WeatherPanel</div>

export default MockWeatherPanel
