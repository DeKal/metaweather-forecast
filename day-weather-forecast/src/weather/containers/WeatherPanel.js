import { connect } from 'react-redux'

import WeatherPanel from 'weather/components/WeatherPanel'
import { getWeathers, getWeatherLoading } from 'state/weather/selectors'

const mapStateToProps = (state) => ({
  weathers: getWeathers(state),
  loading: getWeatherLoading(state)
})

export default connect(mapStateToProps)(WeatherPanel)
