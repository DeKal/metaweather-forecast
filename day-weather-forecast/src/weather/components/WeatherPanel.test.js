import React from 'react'
import { render } from '@testing-library/react'
import WeatherPanel from 'weather/components/WeatherPanel'

const weathers = [
  {
    date: '2021-03-18',
    maxTemperature: 12.4,
    minTemperature: 4.7
  },
  {
    date: '2021-03-19',
    maxTemperature: 13.4,
    minTemperature: 3.7
  },
  {
    date: '2021-03-20',
    maxTemperature: 15.4,
    minTemperature: 4.7
  },
  {
    date: '2021-03-21',
    maxTemperature: 19.4,
    minTemperature: 4.7
  },
  {
    date: '2021-03-22',
    maxTemperature: 12.4,
    minTemperature: 4.7
  },
  {
    date: '2021-03-23',
    maxTemperature: 12.4,
    minTemperature: 4.7
  }
]

describe('WeatherPanel test', () => {
  test('render correctly', () => {
    const { container } = render(<WeatherPanel weathers={weathers} />)
    expect(container).toMatchSnapshot()
  })
  test('render empty panel correctly', () => {
    const { container } = render(<WeatherPanel />)
    expect(container).toMatchSnapshot()
  })
  test('render loading panel correctly', () => {
    const { container } = render(<WeatherPanel loading />)
    expect(container).toMatchSnapshot()
  })
})
