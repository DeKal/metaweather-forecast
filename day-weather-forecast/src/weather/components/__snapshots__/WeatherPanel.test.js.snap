// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherPanel test render correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Thursday 18
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              12°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              5°C
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Friday 19
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              13°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              4°C
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Saturday 20
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              15°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              5°C
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Sunday 21
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              19°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              5°C
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Monday 22
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              12°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              5°C
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Tuesday 23
            </h6>
          </div>
          <div
            class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              12°C
            </h4>
          </div>
          <div
            class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              5°C
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherPanel test render empty panel correctly 1`] = `
<div>
  No weather to display. Please search for a city.
</div>
`;

exports[`WeatherPanel test render loading panel correctly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-container-5"
  >
    <div
      class="MuiCircularProgress-root makeStyles-loading-6 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 36px; height: 36px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
